launch_test.go

```diff
@@ -1,42 +1,156 @@
-		return nil, fmt.Errorf("Error: %s not mocked", cmd)
+	// Working directories should be equal
+	if strings.Compare(wd, wdTest) != 0 {
+		t.Logf("Expected working directory to equal: %s, but was: %s", wd, wdTest)
+		t.FailNow()
 	}
+}
 
-	return []byte(output), nil
+// Test that an invalid shell command will return an error
+func TestShellCommandBadInput(t *testing.T) {
+	_, err := ShellCommand("pwda")
+
+	// Error message should not be nil
+	if err == nil {
+		t.Logf("Expected err to not be: nil")
+		t.Fail()
+	}
 }
 
-func Test(t *testing.T) {
-	shellOut = shellOutMock{}
+// Test that Clone will return without errors and change into the cloned directory
+func TestClone(t *testing.T) {
+	setUp()
+	wd, _ := ShellCommand("pwd")
 
-	shellOutSetup = map[string]string{
-		"/opt/screwdriver/launch.sh abc": "abc",
+	// Arguments
+	GIT_PATH := "github.com/Filbird/pr"
+	GIT_BRANCH := "master"
+
+	err := Clone(GIT_PATH, GIT_BRANCH)
+	// Error message should be nil
+	if err != nil {
+		t.Logf("Expected err to be: nil, but was %v", nil)
+		cleanUp()
+		t.FailNow()
 	}
 
-	output, err := runLauncher("abc")
-	if string(output) != shellOutSetup["/opt/screwdriver/launch.sh abc"] {
-		errmsg := fmt.Sprintf("Expected %s, got %s", shellOutSetup["echo"], output)
-		t.Error(errmsg)
+	wdTest, _ := ShellCommand("pwd")
+	wd = strings.Trim(wd, "\n")
+	wd = wd + "/" + GIT_PATH + "\n"
+	// Working directory should be updated
+	if strings.Compare(wd, wdTest) != 0 {
+		t.Logf("Expected working directory to equal: %s, but was: %s", wd, wdTest)
+		t.Fail()
 	}
-	if err != nil {
-		errmsg := fmt.Sprintf("Error running command: %s", err)
-		t.Error(errmsg)
+
+	cleanUp()
+}
+
+//Test that Clone will return an error if GIT_PATH or GIT_BRANCH is incorrect
+func TestCloneBadInput(t *testing.T) {
+	setUp()
+
+	// Arguments
+	GIT_PATH := "badPath"
+	GIT_BRANCH := "badBranch"
+
+	// Error message should not be nil
+	err := Clone(GIT_PATH, GIT_BRANCH)
+	if err == nil {
+		t.Logf("Expected err to not be: nil")
+		t.Fail()
 	}
+
+	cleanUp()
 }
+
+// // Test that FetchAndMergePR will return without errors and return the correct pull request number
+// func TestFetchAndMergePR(t *testing.T) {
+// 	setUp()
+
+// 	// Arguments
+// 	GIT_BRANCH := "master"
+// 	JOB_NAME := "PR-1"
+
+// 	prTest, err := FetchAndMergePR(GIT_BRANCH, JOB_NAME)
+
+// 	// Error message should be nil
+// 	if err != nil {
+// 		t.Logf("Expected err to be: nil, but was %v", err)
+// 		cleanUp()
+// 		t.FailNow()
+// 	}
+
+// 	pr := "1"
+// 	if strings.Compare(pr, prTest) != 0 {
+// 		t.Logf("Expected pull request number to be: %s, but was %s",pr, prTest)
+// 		cleanUp()
+// 		t.FailNow()
+// 	}
+// }
+
+// Test that ArtifactsDirSetUp will return without errors and exports $ARTIFACTS_DIR
+func TestArtifactsDirSetUp(t *testing.T) {
+	setUp()
+
+	// Call Clone
+	GIT_PATH := "github.com/Filbird/pr"
+	GIT_BRANCH := "master"
+	_ = Clone(GIT_PATH, GIT_BRANCH)
+
+	err := ArtifactsDirSetUp()
+	// Error message should be nil
```

Comment:
This comment doesn't really add value considering that the code below says the same thing